Src/Plugins/AzureDevOpsConfiguration/AZC102_190.DoNotGrantAllPipelinesAccessValidator.cs

```diff
@@ -0,0 +1,154 @@
+        internal const string NotAuthorizedMessage = "Not able to access required ADO API to check.";
+        private const string AdoPatFile = "AdoPat.txt";
+        private static string adoPat = null;
+
+        internal static ValidationState VerifyResponse(string response, ref string message)
+        {
+            PipelinePermission pipelinePermission = JsonConvert.DeserializeObject<PipelinePermission>(response);
+            bool? authorized = pipelinePermission?.AllPipelines?.Authorized;
+            if (authorized != null && authorized.Value == true)
+            {
+                // its shared to all pipelines
+                message = "Was found its accssible to all pipelines.";
+                return ValidationState.Authorized;
+            }
+            else
+            {
+                return ValidationState.NoMatch;
+            }
+        }
+
+        protected override IEnumerable<ValidationResult> IsValidStaticHelper(IDictionary<string, FlexMatch> groups)
+        {
+            if (!groups.TryGetValue("org", out FlexMatch org) ||
+                !groups.TryGetValue("project", out FlexMatch project) ||
+                !groups.TryGetValue("serviceConnectionId", out FlexMatch serviceConnectionId))
+            {
+                return ValidationResult.CreateNoMatch();
+            }
+
+            var validationResult = new ValidationResult
+            {
+                Fingerprint = new Fingerprint
+                {
+                    Host = org.Value.String,
+                    Resource = project.Value.String,
+                    Id = serviceConnectionId.Value.String,
+                    Platform = nameof(AssetPlatform.AzureDevOps),
+                },
+                ValidationState = ValidationState.Unknown,
+            };
+            return new[] { validationResult };
+        }
+
+        protected override ValidationState IsValidDynamicHelper(ref Fingerprint fingerprint,
+                                                                ref string message,
+                                                                IDictionary<string, string> options,
+                                                                ref ResultLevelKind resultLevelKind)
+        {
+            try
+            {
+                string organization = fingerprint.Host;
+                string project = fingerprint.Resource;
+                string serviceConnectionId = fingerprint.Id;
+
+                adoPat ??= ReadPatFromFile(AdoPatFile);
+                if (string.IsNullOrEmpty(adoPat))
+                {
+                    return ValidationState.Unauthorized;
+                }
+
+                HttpClient httpClient = CreateOrRetrieveCachedHttpClient();
+                httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
```

Comment:
>httpClient.DefaultRequestHeaders [](http://example.com/codeflow?start=16&length=32)

create ur own httprequestmessage because the authorization and the accept is restricted to this call only

Reply:
good catch fixed